src/Spryker/Zed/GlossaryStorage/Business/Writer/GlossaryTranslationStorageWriter.php

```diff
@@ -135,7 +135,7 @@ protected function filterInactiveAndEmptyLocalizedStorageEntityTransfers(
             $idGlossaryKey = $glossaryTranslationEntityTransfer->getFkGlossaryKey();
             $localeName = $glossaryTranslationEntityTransfer->getLocale()->getLocaleName();
 
-            if ((!$glossaryTranslationEntityTransfer->getIsActive() || !$glossaryTranslationEntityTransfer->getGlossaryKey()->getIsActive() || !$glossaryTranslationEntityTransfer->getValue())) {
+            if ((!$glossaryTranslationEntityTransfer->getIsActive() || !$glossaryTranslationEntityTransfer->getGlossaryKey()->getIsActive() || trim($glossaryTranslationEntityTransfer->getValue()) === "")) {
```

Comment:
(string) cast would also work instead of trim().

Reply:
Imho it would not have exactly the same effect, as it would not filter strings with multiple whitespaces.

Reply:
Those wouldn't be valid either way IMO.

Reply:
Yes exactly "       ", "  " or "" should not be a valid translation. But with (string) cast a value string "     " would be compared to "" (strict) which would be false. Means it would be treated as valid translation.

Reply:
What I meant is: Such strings are not valid at all. So the translation tool should make sure we don't enter any such whitespace strings into the system.
Your PR is for the scope of `'0'` as valid and literal string. I don't think we should make the scoper bigger and also take care of invalid input such as `' 0'` or `'0 '`, or any combination and amount of leading/trailing whitespace.
Here we can consider then the tooling to be wrong that generated the files we process.

Reply:
Additionally, a trim() with a null input might also not be future proof and PHP8 compatible (as there typehints will require a string only probably).

Reply:
True. So (string) cast would solve the problem the PR was about. NULL should also not be a valid value because it is not allowed in table scheme afaik. But the cast would take care.

I will push another commit.